### DISCUSSION TITLE
How to Pass Dynamic Data from Filament Form to Livewire Component for Price Calculation

### DISCUSSION THREAD

**Discussion by beuzathor** [filamentphp/filament#13797] [Help]
### Package

Form builder

### Package Version

3.2.95

### How can we help you?

I'm trying to create a Livewire component that displays the price of a product within a Filament form. The form allows users to manipulate several select fields (to change color, model, etc.), and each change can affect the price.

My goal is to display the updated price whenever the form values change. I've been using this documentation page as a reference: https://filamentphp.com/docs/3.x/forms/advanced#inserting-livewire-components-into-a-form

Currently, I can pass fixed values to my Livewire component like this:

Forms\Components\Livewire::make(CalculPrixSejour::class, [
    'sejour_id' => '33';
]),

However, it seems impossible to use closures directly. This doesn't work:

Forms\Components\Livewire::make(CalculPrixSejour::class, [
    'sejour_id' => function (Get $get) {
        return $get('sejour_id');
    },
]),

It seems odd that I can't pass anything other than fixed values. I must be missing something in the documentation.

**Comment by tjodalv** [comment]
I am facing exactly the same problem. But interestingly if I convert livewire component to plain Laravel component, then everything works fine. If you do not need livewire reactivity that could be your temporary solution for this issue.

**Comment by gustavo-lucksik** [comment]
Hello guys,

Instead of putting the closure inside the array create a function and return the array with the fields:

```
Livewire::make(CheckOutInfoList::class, function (Get $get) {
     //here you can do some necessary calculation before passing to the component
     return [
          'variable_name1' => $get('field_1'),
          'variable_name2' => $get('field_2'),
          //more fields
     ];
})
```
On component:

```
public $field1;
public $field2;
//others fields

public function mount($variable_name1, $variable_name2, //others variables){
     $this->field1 = $variable_name1;
     $this->field2 = $variable_name2;
     //others variables
}
```

**Reply by vladsimon000000** [reply]
Does not work:
Property [$expeditor_id] not found on component: [app.filament.resources.order-resource.pages.create-order]

When attempting to set public attribute '$expeditor_id' on 'create-order' component is no longer reactive. How are you rendering the attribute in your blade component?

Many thanks.

**Reply by vladsimon000000** [reply]
Many thanks to @leandrocfe over on discord for the solution

```php
Livewire::make(CheckOutInfoList::class, fn (Get $get) => [
    'field_1' => $get('field_1'),
]),
```

```php
namespace App\Livewire;

use Livewire\Attributes\Reactive;
use Livewire\Component;

class CheckOutInfoList extends Component
{
    #[Reactive]
    public ?string $field_1;

    public function render()
    {
        return view('livewire.check-out-info-list');
    }
}
```

```html
<div>
    {{ $field_1 }}
</div>
```

**Comment by ranaabdulrauf** [comment]
isn't that answer your question
from the documentation

```
use Illuminate\Database\Eloquent\Model;
 
class Foo extends Component
{
    public function mount(?Model $record = null): void
    {       
        // ...
    }
    
    // or
    
    public ?Model $record = null;
}
```

**Reply by ranaabdulrauf** [reply]
dd($record) in your livewire component inside mount

**Comment by lucasff** [comment]
I think you can pass a closure for your Component `state()` and from there you return the results of your calculation.
Make sure you have `reactive()` and `live()` defined on your changing fields, then trigger the change using the `afterStateUpdated()`

Hope it helps.

you're passing the closures wrong.
`fn ($get) => CalculPrixSejour::calc($get('sejour_id'))`
`afterStateUpdated(fn ($set, $state) => $set('sejour_id', $state))`
